akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -293,6 +293,51 @@ private[akka] object StreamLayout {
   }
 }
 
+private[stream] object V {
+  sealed trait Termination
+  case object Allowed extends Termination
+  case object Completed extends Termination
+  case class Failed(ex: Throwable) extends Termination
+}
+
+private[stream] final class V[T] extends Processor[T, T] {
+  import V._
+  import ReactiveStreamsCompliance._
+
+  private val subscriptionStatus = new AtomicReference[AnyRef]
+  private val terminationStatus = new AtomicReference[Termination]
+
+  override def subscribe(s: Subscriber[_ >: T]): Unit =
+    if (subscriptionStatus.compareAndSet(null, s)) () // wait for onSubscribe
+    else {
+      val sub = subscriptionStatus.getAndSet(s).asInstanceOf[Subscription]
+      tryOnSubscribe(s, sub)
+      terminationStatus.getAndSet(Allowed) match {
```

Comment:
No, you just need to ignore the null, since setting it to Allowed in the getAndSet will force the onComplete to fail the CAS and call the onComplete itself